### PR TITLE
#54239_test_wp_fuzzy_number_match

### PR DESCRIPTION
**Pull request by pbearne**
Trac ticket: https://core.trac.wordpress.org/ticket/54239#ticket

### CODE REVIEW

File: tests/phpunit/tests/functions/wpFuzzyNumberMatch.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * Test wp_fuzzy_number_match().
+ *
+ * @group functions.php
+ * @covers ::wp_fuzzy_number_match
+ */
+class Tests_wpFuzzyNumberMatch extends WP_UnitTestCase {
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpFuzzyNumberMatch.php]:
```suggestion
class Tests_Functions_WpFuzzyNumberMatch extends WP_UnitTestCase {
```

File: tests/phpunit/tests/functions/wpFuzzyNumberMatch.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+	/**
+	 * @dataProvider data_wp_fuzzy_number_match
+	 *
+	 * @param $expected
+	 * @param $actual
+	 * @param $precision
+	 * @param $result
+	 *
+	 * @return void
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpFuzzyNumberMatch.php:19]:
```suggestion
```
These lines aren't necessary for the docblock.

File: tests/phpunit/tests/functions/wpFuzzyNumberMatch.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+ */
+class Tests_wpFuzzyNumberMatch extends WP_UnitTestCase {
+
+	/**
+	 * @dataProvider data_wp_fuzzy_number_match
+	 *
+	 * @param $expected
+	 * @param $actual
+	 * @param $precision
+	 * @param $result
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpFuzzyNumberMatch.php]: These should have types and descriptions to help testers.

Suggest copying the same entries from [here](https://github.com/WordPress/wordpress-develop/blob/cab053271e272a21b4d140e50ac0d6a78094ecd5/src/wp-includes/functions.php#L8420).

File: tests/phpunit/tests/functions/wpFuzzyNumberMatch.php [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+	 *
+	 * @return void
+	 */
+	public function test_wp_fuzzy_number_match( $expected, $actual, $precision, $result ) {
+		$this->assertEquals( $result, wp_fuzzy_number_match( $expected, $actual, $precision ) );
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpFuzzyNumberMatch.php]: `wp_fuzzy_number_match()` returns `bool`. This test should use `assertSame()`.

File: tests/phpunit/tests/functions/wpFuzzyNumberMatch.php [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+	}
+
+	/**
+	 * @return array[]
+	 */
+	public function data_wp_fuzzy_number_match() {
+
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpFuzzyNumberMatch.php:31]:
```suggestion
```

File: tests/phpunit/tests/functions/wpFuzzyNumberMatch.php [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+	 * @param $result
+	 *
+	 * @return void
+	 */
+	public function test_wp_fuzzy_number_match( $expected, $actual, $precision, $result ) {
+		$this->assertEquals( $result, wp_fuzzy_number_match( $expected, $actual, $precision ) );
+	}
+
+	/**
+	 * @return array[]
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpFuzzyNumberMatch.php:28]:
```suggestion
	 * Data provider.
	 *
	 * @return array[]
```

File: tests/phpunit/tests/functions/wpFuzzyNumberMatch.php [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+	 */
+	public function data_wp_fuzzy_number_match() {
+
+		return array(
+			1            => array(
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpFuzzyNumberMatch.php]: The first three keys should be enclosed within single quotes and would benefit from the type being included for consistency with the other datasets:

```
1 should be '1 int'
2 should be '2 int'
3 should be '3 int'
```

File: tests/phpunit/tests/functions/wpFuzzyNumberMatch.php [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+				'actual'    => 11,
+				'precision' => 10,
+				'result'    => true,
+			),
+			'12 with 10' => array(
+				'expected'  => 1,
+				'actual'    => 12,
+				'precision' => 10,
+				'result'    => false,
+			),
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpFuzzyNumberMatch.php:68]: `wp_fuzzy_number_match()` accepts `int|float` for the `$expected` and `$actual` values.

This data provider would benefit from some additional datasets to test float values.

File: tests/phpunit/tests/functions/wpFuzzyNumberMatch.php [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+ * Test wp_fuzzy_number_match().
+ *
+ * @group functions.php
+ * @covers ::wp_fuzzy_number_match
+ */
+class Tests_wpFuzzyNumberMatch extends WP_UnitTestCase {
+
+	/**
+	 * @dataProvider data_wp_fuzzy_number_match
+	 *
```
Comment (costdev) [inline tests/phpunit/tests/functions/wpFuzzyNumberMatch.php:13]:
```suggestion
	 *
	 * @ticket 54239
	 *
```

**Review by costdev — CHANGES_REQUESTED** [review]
Thanks for these tests @pbearne!

This review suggests changes for:
- Consistency in the test suite.
- Stricter assertions.
- Ensuring that dataset keys show correctly when using the `--testdox` parameter.
- More datasets.

File: tests/phpunit/tests/functions/wpFuzzyNumberMatch.php (hunk 8 above)
Comment (pbearne) [reply tests/phpunit/tests/functions/wpFuzzyNumberMatch.php:68]: added

**Pull request closed without merging**